### PR TITLE
Feature/send branch to healthchecks

### PR DESCRIPTION
**Pull request by PierreDelpy**
added some refactoring to define the directories once and more importantly:
added monitoring for branches (lines 112-116)

### CODE REVIEW

**Review by Threated — APPROVED** [review]

File: lib/prerequisites.sh [hunk 1]
```diff
@@ -32,39 +34,39 @@ fi
 log INFO "Checking configuration ..."
 
 ## Download submodule
-if [ ! -d "/etc/bridgehead/" ]; then
-  fail_and_report 1 "Please set up the config folder at /etc/bridgehead. Instruction are in the readme."
+if [ ! -d ${configDirectory} ]; then
```
Comment (TKussel) [inline lib/prerequisites.sh:37]:
```suggestion
if [ ! -d "${configDirectory}" ]; then
```

File: lib/prerequisites.sh [hunk 2]
```diff
@@ -3,14 +3,16 @@
...
 
 if ! id "bridgehead" &>/dev/null; then
   log ERROR "User bridgehead does not exist. Please run bridgehead install $PROJECT"
   exit 1
 fi
 
-checkOwner /srv/docker/bridgehead bridgehead || exit 1
-checkOwner /etc/bridgehead bridgehead || exit 1
+checkOwner ${configDirectory} bridgehead || exit 1
+checkOwner ${componentDirectory} bridgehead || exit 1
```
Comment (TKussel) [inline lib/prerequisites.sh:15]:
```suggestion
checkOwner "${configDirectory}" bridgehead || exit 1
checkOwner "${componentDirectory}" bridgehead || exit 1
```

File: lib/prerequisites.sh [hunk 3]
```diff
@@ -32,39 +34,39 @@ fi
...
   mkdir -p /etc/bridgehead/traefik-tls
 fi
 
-if [ ! -e "/etc/bridgehead/traefik-tls/fullchain.pem" ]; then
+if [ ! -e "${configDirectory}traefik-tls/fullchain.pem" ]; then
   openssl req -x509 -newkey rsa:4096 -nodes -keyout /etc/bridgehead/traefik-tls/privkey.pem -out /etc/bridgehead/traefik-tls/fullchain.pem -days 3650 -subj "/CN=$HOST"
 fi
 
-if [ -e /etc/bridgehead/vault.conf ]; then
+if [ -e $configDirectory}vault.conf ]; then
```
Comment (TKussel) [inline lib/prerequisites.sh:61]:
```suggestion
if [ -e "${configDirectory}"vault.conf ]; then
```

Comment (TKussel) [reply lib/prerequisites.sh:61]: Note the missing opening brace

File: lib/prerequisites.sh [hunk 4]
```diff
@@ -32,39 +34,39 @@ fi
...
   if [ "$(stat -c "%a %U" /etc/bridgehead/vault.conf)" != "600 bridgehead" ]; then
     fail_and_report 1 "/etc/bridgehead/vault.conf has wrong owner/permissions. To correct this issue, run chmod 600 /etc/bridgehead/vault.conf && chown bridgehead /etc/bridgehead/vault.conf."
   fi
 fi
 
 log INFO "Checking network access ($BROKER_URL_FOR_PREREQ) ..."
 
-source /etc/bridgehead/${PROJECT}.conf
+source ${configDirectory}${PROJECT}.conf
```
Comment (TKussel) [inline lib/prerequisites.sh:69]:
```suggestion
source "${configDirectory}${PROJECT}".conf
```

File: lib/prerequisites.sh [hunk 5]
```diff
@@ -92,10 +94,10 @@ if [ "${PROJECT}" != "minimal" ]; then
   fi
 fi
 checkPrivKey() {
-  if [ -e /etc/bridgehead/pki/${SITE_ID}.priv.pem ]; then
+  if [ -e ${configDirectory}pki/${SITE_ID}.priv.pem ]; then
```
Comment (TKussel) [inline lib/prerequisites.sh:97]:
```suggestion
  if [ -e "${configDirectory}"pki/"${SITE_ID}".priv.pem ]; then
```

File: lib/prerequisites.sh [hunk 6]
```diff
@@ -107,6 +109,11 @@ else
   checkPrivKey || exit 1
 fi
 
+for dir in  ${configDirectory} ${componentDirectory}; do
```
Comment (TKussel) [inline lib/prerequisites.sh:112]:
```suggestion
for dir in  "${configDirectory}" "${componentDirectory}"; do
```

File: lib/prerequisites.sh [hunk 7]
```diff
@@ -3,14 +3,16 @@
 source lib/functions.sh
 
 detectCompose
+configDirectory="/etc/bridgehead/"
+componentDirectory="/srv/docker/bridgehead/"
```
Comment (TKussel) [inline lib/prerequisites.sh:7]: Is the convention to use camel case variables for local variables? Otherwise, please convert to upper case

**Review by TKussel — CHANGES_REQUESTED** [review]

**Review by patrickskowronekdkfz — CHANGES_REQUESTED** [review]
I like the Idea of having the option to move the bridgehead in other directories. However, I would strongly advise to check if the path is valid. Also if you would change the path, this needs to be considered in the systemd files and the other scripts.

File: lib/prerequisites.sh (hunk 7 above)
Comment (PierreDelpy) [reply lib/prerequisites.sh]: Thx for checking, added your suggestions

**Comment by TKussel** [comment]
> I like the Idea of having the option to move the bridgehead in other directories. However, I would strongly advise to check if the path is valid. Also if you would change the path, this needs to be considered in the systemd files and the other scripts.

The config dir is explixitly checked. How does the "check ownership" function behave with non existing paths? Maybe this already handles the component dir.

I think this is a sensible change, even if systemd is not yet fully adapted (could be done by extracting the variables into a separate file and loding that as an env-file in the systemd service). Could that be done in a future PR?

**Review by TKussel — APPROVED** [review]

**Review by patrickskowronekdkfz — APPROVED** [review]

**Comment by patrickskowronekdkfz** [comment]
> > I like the Idea of having the option to move the bridgehead in other directories. However, I would strongly advise to check if the path is valid. Also if you would change the path, this needs to be considered in the systemd files and the other scripts.
> 
> The config dir is explixitly checked. How does the "check ownership" function behave with non existing paths? Maybe this already handles the component dir.
> 
> I think this is a sensible change, even if systemd is not yet fully adapted (could be done by extracting the variables into a separate file and loding that as an env-file in the systemd service). Could that be done in a future PR?

Yes, I see it as a future PR.

File: lib/prerequisites.sh [hunk 8]
```diff
@@ -92,10 +94,10 @@ if [ "${PROJECT}" != "minimal" ]; then
   fi
 fi
 checkPrivKey() {
-  if [ -e /etc/bridgehead/pki/${SITE_ID}.priv.pem ]; then
+  if [ -e "${CONFIG_DIR}"pki/"${SITE_ID}".priv.pem ]; then
```
Comment (Threated) [inline lib/prerequisites.sh:97]: This should be like so no?
```suggestion
  if [ -e "${CONFIG_DIR}pki/${SITE_ID}.priv.pem" ]; then
```

**Review by Threated — CHANGES_REQUESTED** [review]

File: lib/prerequisites.sh (hunk 8 above)
Comment (TKussel) [reply lib/prerequisites.sh:97]: It should expand to the same thing, but is the cleaner solution. I agree.

**Review by Threated — APPROVED** [review]

**Pull request merged**